llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -10225,8 +10225,10 @@ SDValue TargetLowering::LowerToTLSEmulatedModel(const GlobalAddressSDNode *GA,
 
   ArgListTy Args;
   ArgListEntry Entry;
-  std::string NameString = ("__emutls_v." + GA->getGlobal()->getName()).str();
-  Module *VariableModule = const_cast<Module*>(GA->getGlobal()->getParent());
+  const GlobalValue *GV = GA->getGlobal();
+  GV = GV->getAliaseeObject() ? GV->getAliaseeObject() : GV;
+  std::string NameString = ("__emutls_v." + GV->getName()).str();
```

Comment:
SmallString

Reply:
The original code used 'NameString'. And the lenth of rvalue at line 10230 is indeterminate. So I don't think it's a good idea to use 'SmallString'.

Reply:
It works for any size of string, it's just optimized for the common case where the name is small

Reply:
done